### ISSUE TITLE
[indent] expected indentation inconsistent for `class X extends/implements Y`

### ISSUE THREAD

**Issue by quezak** [typescript-eslint/typescript-eslint#1597]
_intro note: I understand formatting line breaks & indents in a multi-line declaration like `class X extends Y implements I1, I2, I3` is a complicated & opinionated problem, especially as the class names grow longer and/or have generic args (~~inb4 "use prettier" :sweat_smile:~~), I'm just reporting a small inconsistency between how `extends` and `implements` behaves._

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/indent": ["error", 4]
  }
}
```

```TS
// ---- example A ----
export class ExA1
implements Iface1 {  // OK, indent 0 is correct
    fn(): void {}
}

export class ExA2
extends Base2 {  // Error: Expected indentation of 4 spaces but found 0.
    fn(): void {}
}

// ---- example B ----
export class ExB1 implements
Iface1 {  // OK, indent 0 is correct
    fn(): void {}
}

export class ExB2 extends
Base2 {  // Error: Expected indentation of 4 spaces but found 0.
    fn(): void {}
}
```

**Expected Result**
In the above examples, `... extends X ...` and `... implements Y ...` with identical line breaks should have identical indentation.

**Actual Result**
`extends` and `implements` require different indent when the class declaration is multi-line.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.19.2` |
| `@typescript-eslint/parser`        | `2.19.2` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `11.9` |
| `npm`                              | `6.5` |

(but I don't use `npm`, only `yarn` ^)

**Comment by bradzacher** [comment]
Merging into #1824
